ambry-api/src/main/java/com.github.ambry/messageformat/BlobProperties.java

```diff
@@ -69,8 +70,13 @@ public BlobProperties(long blobSize, String serviceId, String ownerId, String co
     this.ownerId = ownerId;
     this.contentType = contentType;
     this.isPrivate = isPrivate;
-    this.timeToLiveInSeconds = timeToLiveInSeconds;
     this.creationTimeInMs = creationTimeInMs;
+    long maxAllowedTTLInMs = Utils.maxEpochTimeInMs - creationTimeInMs;
+    if (TimeUnit.SECONDS.toMillis(timeToLiveInSeconds) > maxAllowedTTLInMs) {
+      this.timeToLiveInSeconds = Utils.Infinite_Time;
```

Comment:
so the decision is to `Utils.Infinite` as opposed to setting this to `maxAllowedTTLinMs`?